test/test_unary_ufuncs.py

```diff
@@ -1186,34 +1186,46 @@ def test_i0_special(self, device, dtype):
     @dtypesIfCUDA(*torch.testing.get_all_fp_dtypes())
     @dtypes(torch.bfloat16, torch.float32, torch.float64)
     @unittest.skipIf(not TEST_SCIPY, "SciPy not found")
-    def test_special_i0e_vs_scipy(self, device, dtype):
-        def check_equal(t):
+    def test_special_i0_i1_vs_scipy(self, device, dtype):
```

Comment:
How would we need to change the reference numerics tests so that we can delete this operation?

Reply:
Near the following line
https://github.com/pytorch/pytorch/blob/71f4c5c1f436258adc303b710efb3f41b2d50c4e/test/test_unary_ufuncs.py#L183 

 we can probably do,
```python
info = torch.finfo(dtype)
vals = _float_extremals  + [info.min, info.max, info.eps, info.tiny]
```

However we will still miss out on the `range` test
https://github.com/pytorch/pytorch/blob/71f4c5c1f436258adc303b710efb3f41b2d50c4e/test/test_unary_ufuncs.py#L1209-L1214

can we add this range based test as an opt-in test for unary-ufuncs?

Reply:
I think I'm missing something here: should we be randomly generating values within that range as part of test_reference_numerics_hard?

Reply:
Separately: expanding the "extremal" values as you suggest sounds like a good idea, would you take a note about that?

Reply:
> Separately: expanding the "extremal" values as you suggest sounds like a good idea, would you take a note about that?

Already done :)

https://github.com/pytorch/pytorch/blob/e34f285e586809eb0e7a2f90d348c578fe7a0de7/test/test_unary_ufuncs.py#L183-L184

Tested locally and the tests passed. Waiting for green signal on CI

Reply:
> I think I'm missing something here: shouldn't we be randomly generating values within that range as part of test_reference_numerics_hard?

Right now for `hard`
We check only for the following values and not randomly sampled,
https://github.com/pytorch/pytorch/blob/71f4c5c1f436258adc303b710efb3f41b2d50c4e/test/test_unary_ufuncs.py#L59-L62
https://github.com/pytorch/pytorch/blob/71f4c5c1f436258adc303b710efb3f41b2d50c4e/test/test_unary_ufuncs.py#L154-L165

Reply:
Yep; that's makes sense. We should extend make_tensor() to compute values beyond -9 to 9 and change the "hard" test to use a wider range.

Reply:
Noted. Will follow-up on this and extremal extension using finfo in a separate PR.